content/kb/troubleshooting-some-old-instance-manager-pods-are-still-running-after-upgrade.md

```diff
@@ -0,0 +1,79 @@
+---
+title: "Troubleshooting: Some old instance manager pods are still running after upgrade"
+author: Derek Su
+draft: false
+date: 2021-11-09
+categories:
+  - "Longhorn Upgrade"
+---
+
+## Applicable versions
+
+Longhorn >= v0.8.0.
+
+## Symptoms
+
+Some old instance manager pods are still running after upgrade.
+
+## Details
+
+**In the following paragraphs, we will explain why some old instance manager pods are still running after upgrade is an expected behavior rather than a bug.**
+
+Let us first take a look of the example. We created a pod with a volume backed by 3 replicas in a Kubernetes cluster with 1 master and 4 workers nodes. The running volume is associated with the engine instance manager pod `instance-manager-e-ec3eb207`.
+
+- Before upgrade, what the current instance manager pods are and which node the volume is on:
+```
+# kubectl -n longhorn-system get pods --sort-by=.metadata.name -o go-template='{{range .items}}{{.metadata.name}} {{.spec.nodeName}}{{"\n"}}{{end}}' | grep instance-manager
+
+instance-manager-e-1ce9451d worker4
+instance-manager-e-7a505ed8 worker2
+instance-manager-e-9e3a3735 worker1
+instance-manager-e-11d7bb9b worker3
+instance-manager-e-ec3eb207 master
+instance-manager-r-6ee37472 master
+instance-manager-r-38dd4e09 worker3
+instance-manager-r-476c588c worker1
+instance-manager-r-89006932 worker2
+instance-manager-r-fe73789e worker4
+
+# kubectl -n longhorn-system get lhv
+NAME                                       STATE      ROBUSTNESS   SCHEDULED   SIZE         NODE          AGE
+pvc-6d229727-ab78-4a72-a4db-9a3279086935   attached   healthy      True        2147483648   master        41m
+```
+
+- After upgrading the longhorn-manager and longhorn-engine of the running volume, what the current instance manager pods are:
```

Comment:
```
After upgrading the Longhorn system then applying engine image upgrade for the running volume, all old instance manager pods except for `instance-manager-e-ec3eb207` are gone.
```

- To avoid confusion, do not say `upgrading the longhorn-manager`.
- Directly summarize the result of the upgrade.